pytorch_lightning/strategies/fully_sharded.py

```diff
@@ -119,32 +132,53 @@ def __init__(
         self.min_num_params = min_num_params
         self.state_dict_device = torch.device("cpu") if state_dict_to_cpu else None
         self._process_group = None
+        self._rank_0_will_call_children_scripts: bool = False
```

Comment:
> Where did _share_information_to_prevent_deadlock go?

That's right, since its not called in the setup method, its not needed in this file AFAICT.